Web Inspector: REGRESSION(252653@main): Elements: clicking a layout badge doesn't do anything

dcrousso opened the pull request:
#### f8a09808de412899ef3aa2f23386118e3546b2ec
<pre>
Web Inspector: REGRESSION(252653@main): Elements: clicking a layout badge doesn&apos;t do anything
<a href="https://bugs.webkit.org/show_bug.cgi?id=243552">https://bugs.webkit.org/show_bug.cgi?id=243552</a>

Reviewed by Patrick Angle.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
(WI.DOMTreeElement.prototype._updateLayoutBadge):
(WI.DOMTreeElement.prototype._handleLayoutFlagsChanged):
The event listeners for the layout overlay being shown/hidden exist to update the layout type badge.
Instead of adding/removing those event listeners whenever the layout flags change, do it when the
layout badge is created/destroyed as that&apos;s really what needs it, not the rest of `WI.DOMTreeElement`.
This also simplifies the logic for when we add/remove those event listeners.

Canonical link: <a href="https://commits.webkit.org/253219@main">https://commits.webkit.org/253219@main</a>
</pre>

patrickangle commented on Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
```diff
@@ -2041,6 +2048,11 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
+        if (!hadLayoutBadge) {
+            this.representedObject.addEventListener(WI.DOMNode.Event.LayoutOverlayShown, this._updateLayoutBadgeStatus, this);
+            this.representedObject.addEventListener(WI.DOMNode.Event.LayoutOverlayHidden, this._updateLayoutBadgeStatus, this);
+        }
```
Can this be an `else` off of :2022-2028 instead? It would be clearer IMO if the adding/removing of the layout event listeners were together such that one or the other happens. This function overall is very hard to grok, and bringing those calls closer together will make it easier to do so.

dcrousso replied:
i cant make it an `else` as that would go against the styleguide (since the `if` has a `return`), but i can move this up so that they're closer together :)

dcrousso committed:
Web Inspector: REGRESSION(252653@main): Elements: clicking a layout badge doesn't do anything
https://bugs.webkit.org/show_bug.cgi?id=243552

Reviewed by Patrick Angle.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js:
(WI.DOMTreeElement.prototype._updateLayoutBadge):
(WI.DOMTreeElement.prototype._handleLayoutFlagsChanged):
The event listeners for the layout overlay being shown/hidden exist to update the layout type badge.
Instead of adding/removing those event listeners whenever the layout flags change, do it when the
layout badge is created/destroyed as that's really what needs it, not the rest of `WI.DOMTreeElement`.
This also simplifies the logic for when we add/remove those event listeners.

Canonical link: https://commits.webkit.org/253219@main
